rainbow-me/rainbow
# 🏛 Migrate `useENSAddress` to Async State RFC

Fixes RNBW-4505

## What changed (plus any additional context for devs)

This PR migrates the `useENSAddress` hook to the new Async State architecture, following the template used in #4164.


## Screen recordings / screenshots

N/A


## What to test

- Ensure the "View ENS Profile" sheet still works,
- Ensure the "Edit ENS Profile" sheet still works,
- Ensure opening ENS NFTs still work,
- Ensure searching for ENS profiles still works.
